### DISCUSSION TITLE
Allow title field to be optional? [5.0+]

### DISCUSSION THREAD

**Discussion by dlindberg** [craftcms/cms#15535] [Ideas] [answered]
I've run into a few instances where it would be helpful to both enable the title field on an entry type and allow it to be optional. The specific cases is when setting up an entry type to handle figures (images, tables, etc.) where they might or might not have / need a title.

The built in title field is currently always required (though it doesn't flag as required when working on the entry type settings) whenever it is enabled. So in cases where I want an optional title field I end up creating an additional plain text title field and allowing it to be optional.

I think lack of the required flag when enabled in the field editor is probably a bug. And for entry types that aren't being used in sections I think it would be helpful if you could set it as optional like other fields.

I'm not thinking of a clear reason why using an extra "title like" field in cases where it is optional creates a benefit, but also don't necessarily think it is a significant problem.

**Comment by brandonkelly** [comment]
You could pull this off by adding a custom “Title” Plain Text field to the layout, disable entry type’s “Show the Title field” setting, and set the Title Format to pull in the custom field’s value if set, and fallback on something else if empty.

**Reply by dlindberg** [reply]
Sure, it's not a significant issue, but it is also kind of annoying. You end up with a stray faux title field you don't really need and that you have some extra steps if you want to template it as title or you can just use the faux title directly as long as you don't actually call it title. It ends up feeling like a loose end that future developers would need to be aware of, but yeah it's not really a user facing issue.

**Reply by brandonkelly** [reply]
True… reopened and we can consider this down the road.

**Comment by brandonkelly** [comment, marked as answer]
Just added the ability to make Title fields optional, for Craft 5.5 (#15942) 🎉

**Reply by brandonkelly** [reply]
Craft 5.5.0 is out with that change 🎉
